xlators/storage/posix/src/posix-handle.c

```diff
@@ -410,58 +392,54 @@ posix_handle_path(xlator_t *this, uuid_t gfid, const char *basename, char *ubuf,
     char *base_str = NULL;
     int base_len = 0;
     int pfx_len;
-    int maxlen;
+    int maxlen = PATH_MAX;
     char *buf;
     int index = 0;
     int dfd = 0;
-    char newstr[POSIX_GFID_HASH2_LEN] = {
-        0,
-    };
+    char newstr[POSIX_GFID_HASH2_LEN];
 
     priv = this->private;
 
     uuid_str = uuid_utoa(gfid);
 
     if (ubuf) {
         buf = ubuf;
-        maxlen = size;
     } else {
-        maxlen = PATH_MAX;
         buf = alloca(maxlen);
     }
 
     index = gfid[0];
     dfd = priv->arrdfd[index];
 
-    base_len = (priv->base_path_length + SLEN(GF_HIDDEN_PATH) + 45);
+    base_len = (priv->base_path_length + SLEN(GF_HIDDEN_PATH) +
+                POSIX_GFID_HASH2_LEN);
     base_str = alloca(base_len + 1);
-    base_len = snprintf(base_str, base_len + 1, "%s/%s/%02x/%02x/%s",
-                        priv->base_path, GF_HIDDEN_PATH, gfid[0], gfid[1],
-                        uuid_str);
-    pfx_len = priv->base_path_length + 1 + SLEN(GF_HIDDEN_PATH) + 1;
+    snprintf(newstr, sizeof(newstr), "%02x/%s", gfid[1], uuid_str);
+    base_len = snprintf(base_str, base_len + 1, "%s/%s/%02x/%s",
+                        priv->base_path, GF_HIDDEN_PATH, gfid[0], newstr);
```

Comment:
We can avoid doing two `snprintf()` and `newstr` can be a pointer instead of an array:

```suggestion
    base_len = snprintf(base_str, base_len + 1, "%s/%s/%02x/%02x/%s",
                        priv->base_path, GF_HIDDEN_PATH, gfid[0], gfid[1],
                        uuid_str);
    newstr = base_str + base_len - UUID_CANONICAL_FORM_LEN - 3;
```